src/basic/chase-symlinks.c

```diff
@@ -380,13 +376,10 @@ int chase_symlinks_at(
         if (ret_path)
                 *ret_path = TAKE_PTR(done);
 
-        if (ret_fd) {
+        if (ret_fd)
                 /* Return the O_PATH fd we currently are looking to the caller. It can translate it to a
                  * proper fd by opening /proc/self/fd/xyz. */
-
-                assert(fd >= 0);
-                *ret_fd = TAKE_FD(fd);
-        }
+                *ret_fd = exists ? TAKE_FD(fd) : -EBADF;
```

Comment:
please add a test case that ensures this always returns the expected fd

Reply:
hmm, if you chase a path like /a/b/c/d and /a/b exist, but /a/b/c does not, then this will return an fd to /a/b, right? But if /a/b/c exists but /a/b/c/d does not, then it will return an fd to /a/b/c, right?

So this basically means depending on what exists and what does not you might get an fd back that referes to completely different dirs, and you simply don't know to which one because we don#t return any info about that

Reply:
`ret_fd` is only set if `exists` is `true`. If any component in the path does not exist, `exists` is set to `false` which means `ret_fd` will be set to `-EBADF`. In other words, `ret_fd` is only initialized if the resolved path actually exists so I'm not sure what the problem is here?